src/picongpu/include/plugins/particleDetector/ParticleDetector.hpp

```diff
@@ -0,0 +1,542 @@
+    float3_X detectorFrameVecY;
+    float3_X detectorFrameVecZ;
+
+    typedef PMacc::container::DeviceBuffer<float_X, DIM3> DBufCalorimeter;
+    typedef PMacc::container::HostBuffer<float_X, DIM3> HBufCalorimeter;
+
+    /* device calorimeter buffer for a single gpu */
+    DBufCalorimeter* dBufCalorimeter;
+    /* device calorimeter buffer for all particles which have left the simulation volume  */
+    DBufCalorimeter* dBufLeftParsCalorimeter;
+    /* host calorimeter buffer for a single mpi rank */
+    HBufCalorimeter* hBufCalorimeter;
+    /* host calorimeter buffer for summation of all mpi ranks */
+    HBufCalorimeter* hBufTotalCalorimeter;
+
+public:
+    typedef CalorimeterFunctor<typename DBufCalorimeter::Cursor> MyCalorimeterFunctor;
+private:
+    typedef boost::shared_ptr<MyCalorimeterFunctor> MyCalorimeterFunctorPtr;
+    MyCalorimeterFunctorPtr calorimeterFunctor;
+
+    typedef boost::shared_ptr<PMacc::algorithm::mpi::Reduce<simDim> > AllGPU_reduce;
+    AllGPU_reduce allGPU_reduce;
+
+    void restart(uint32_t restartStep, const std::string restartDirectory)
+    {
+        if(this->notifyPeriod == 0)
+            return;
+
+        HBufCalorimeter hBufLeftParsCalorimeter(this->dBufLeftParsCalorimeter->size());
+
+        int rank;
+        MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+        if(rank == 0)
+        {
+            splash::SerialDataCollector hdf5DataFile(1);
+            splash::DataCollector::FileCreationAttr fAttr;
+
+            splash::DataCollector::initFileCreationAttr(fAttr);
+            fAttr.fileAccType = splash::DataCollector::FAT_READ;
+
+            std::stringstream filename;
+            filename << restartDirectory << "/" << prefix << restartStep;
+
+            hdf5DataFile.open(filename.str().c_str(), fAttr);
+
+            splash::Dimensions dimensions;
+
+            hdf5DataFile.read(restartStep,
+                              "detectorLeftPars",
+                              dimensions,
+                              &(*hBufLeftParsCalorimeter.origin()));
+
+            hdf5DataFile.close();
+
+            /* rank 0 divides and distributes the detector to all ranks in equal parts */
+            int numRanks;
+            MPI_Comm_size(MPI_COMM_WORLD, &numRanks);
+            using namespace lambda;
+            PMacc::algorithm::host::Foreach()(hBufLeftParsCalorimeter.zone(),
+                                              hBufLeftParsCalorimeter.origin(),
+                                              _1 = _1 / float_X(numRanks));
+        }
+
+        MPI_Bcast(&(*hBufLeftParsCalorimeter.origin()),
+                  hBufLeftParsCalorimeter.size().productOfComponents() * sizeof(float_X),
+                  MPI_CHAR,
+                  0,
+                  MPI_COMM_WORLD);
+
+        *this->dBufLeftParsCalorimeter = hBufLeftParsCalorimeter;
+    }
+
+
+    void checkpoint(uint32_t currentStep, const std::string checkpointDirectory)
+    {
+        if(this->notifyPeriod == 0)
```

Comment:
Is it essential that your checkpoint period is equal to your data output period? If not, please only write backups at checkpoint intervals.